src/main/java/org/fcrepo/storage/ocfl/DefaultOcflObjectSessionFactory.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.storage.ocfl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.wisc.library.ocfl.api.OcflRepository;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Default OcflObjectSessionFactory implementation
+ *
+ * @author pwinckles
+ */
+public class DefaultOcflObjectSessionFactory implements OcflObjectSessionFactory {
+
+    private final OcflRepository ocflRepo;
+    private final Path stagingRoot;
+    private final ObjectMapper objectMapper;
+    private final String defaultVersionMessage;
+    private final String defaultVersionUserName;
+    private final String defaultVersionUserAddress;
+
+    private final Map<String, OcflObjectSession> sessions;
+
+    public DefaultOcflObjectSessionFactory(final OcflRepository ocflRepo,
+                                           final Path stagingRoot,
+                                           final ObjectMapper objectMapper,
+                                           final String defaultVersionMessage,
+                                           final String defaultVersionUserName,
+                                           final String defaultVersionUserAddress) {
+        this.ocflRepo = ocflRepo;
+        this.stagingRoot = stagingRoot;
+        this.objectMapper = objectMapper;
```

Comment:
Where is the objectMapper going to be initialized? My understanding is that ObjectMapper is threadsafe, but I'm a little unclear from the comments in this thread (particularly the second response) whether a global instance of it can result in some blocking issues:
https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field

Reply:
It's fairly standard to use a global ObjectMapper instance. Frameworks like Spring Boot provide one by default. Regardless, this class doesn't care if it's a global instance of not. It's just exposed for DI.